integration-test/src/test/java/uk/gov/gchq/gaffer/integration/junit/extensions/IntegrationTestSuiteExtensionTest.java

```diff
@@ -0,0 +1,380 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.integration.junit.extensions;
+
+import org.apache.commons.collections4.MapUtils;
+import org.apache.commons.collections4.SetUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.platform.suite.api.SelectClasses;
+import org.junit.platform.suite.api.Suite;
+import org.junit.platform.testkit.engine.EngineTestKit;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.fail;
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+import static org.junit.platform.testkit.engine.EventConditions.event;
+import static org.junit.platform.testkit.engine.EventConditions.finishedSuccessfully;
+import static org.junit.platform.testkit.engine.EventConditions.finishedWithFailure;
+import static org.junit.platform.testkit.engine.EventConditions.skippedWithReason;
+import static org.junit.platform.testkit.engine.EventConditions.test;
+import static org.junit.platform.testkit.engine.EventConditions.uniqueIdSubstring;
+import static org.junit.platform.testkit.engine.TestExecutionResultConditions.message;
+import static uk.gov.gchq.gaffer.integration.junit.extensions.IntegrationTestSuiteExtension.INIT_CLASS;
+
+class IntegrationTestSuiteExtensionTest {
+
+    public static final String JUNIT_PLATFORM_SUITE = "junit-platform-suite";
+
+    public static final String TEST_STRING = "test";
+
+    public static final String TEST_STRING_2 = "test 2";
+
+    public static final Integer TEST_INTEGER = 1234;
+
+    private static final String ERROR_THE_TEST_SHOULD_RUN = "The test should not be run as the IntegrationTestSuiteExtension fails first as no initClass was set";
+
+    @SelectClasses(value = {MethodInjectionIT.class})
+    static class MethodInjectionTestSuite extends AbstractTestSuite {
+        @Override
+        public Optional<Set<Object>> getObjects() {
+            return Optional.of(SetUtils.unmodifiableSet(TEST_STRING, TEST_INTEGER));
+        }
+    }
+
+    @ExtendWith(IntegrationTestSuiteExtension.class)
+    static class MethodInjectionIT {
+        @Test
+        void shouldSucceedMethodInjectionString(@IntegrationTestSuiteInstance final String actualString) {
+            assertThat(actualString).isEqualTo(TEST_STRING);
+        }
+
+        @Test
+        void shouldSucceedMethodInjectionInteger(@IntegrationTestSuiteInstance final Integer actualInteger) {
+            assertThat(actualInteger).isEqualTo(TEST_INTEGER);
+        }
+    }
+
+    @Test
+    void shouldSucceedMethodInjectionTest() {
```

Comment:
Specifically, it is line 277 which is not tested according the jacoco results. As you say, it does look as though this is tested and having had a closer look, it seems this is a false positive and this is actually tested. I inserted a logging message in the catch and that is shown as being covered, but not `ExceptionUtils.throwAsUncheckedException(t);`. 

I imagine this is something to do with how it handles the exception which causes jacoco to think it is not covered. What's the reason for using this? [Looking at the docs](https://junit.org/junit5/docs/5.2.0/api/index.html?org/junit/platform/commons/util/ExceptionUtils.html), I think we ought to avoid it if possible. We can use `throw new RuntimeException(t)` instead to convert to an unchecked exception.

Reply:
I got the code from another extension when finding out how to inject. I can rework. Seems that all the `ExceptionUtils.throwAsUncheckedException(t)` does is [re-throw](https://github.com/junit-team/junit5/blob/3a5da52a0b3f5af5aeb9d9914643b1274d3f854e/junit-platform-commons/src/main/java/org/junit/platform/commons/util/ExceptionUtils.java#L58) the same `Throwable` anyhow...

Reply:
This has now been replaced by throwing a RuntimeException instead - which shows as all being covered. Fixed.